peer.go

```diff
@@ -76,9 +76,20 @@ func newPeerList(root *RootPeerList) *PeerList {
 	}
 }
 
-// SetStrategy sets customized peer selection stratedgy.
+// SetStrategy sets customized peer selection strategy.
 func (l *PeerList) SetStrategy(sc ScoreCalculator) {
+	l.Lock()
+	defer l.Unlock()
 	l.scoreCalculator = sc
```

Comment:
It currently gets an rlock to get the current score and to check whether the peer exists. Perhaps that same function should return the score calculator?